geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelTestBase.java

```diff
@@ -132,380 +124,227 @@ public final void postSetUp() throws Exception {
-          return cache.getCacheServers().size() == 1;
-        }
+    await().atMost(180, SECONDS).pollInterval(1, SECONDS)
+        .until(() -> cache.getCacheServers().size(), equalTo(1));
 
-        public String description() {
-          return excuse;
-        }
-      };
-      Wait.waitForCriterion(wc, 3 * 60 * 1000, 1000, true);
-
-      CacheServerImpl bs = (CacheServerImpl) cache.getCacheServers().iterator().next();
-      assertNotNull(bs);
-      assertNotNull(bs.getAcceptor());
-      assertNotNull(bs.getAcceptor().getCacheClientNotifier());
-      final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
-      wc = new WaitCriterion() {
-        String excuse;
-
-        public boolean done() {
-          return ccn.getClientProxies().size() > 0;
-        }
+    CacheServerImpl bs = (CacheServerImpl) cache.getCacheServers().iterator().next();
+    assertThat(bs).isNotNull();
+    assertThat(bs.getAcceptor()).isNotNull();
+    assertThat(bs.getAcceptor().getCacheClientNotifier()).isNotNull();
 
-        public String description() {
-          return excuse;
-        }
-      };
-      Wait.waitForCriterion(wc, 3 * 60 * 1000, 1000, true);
-
-      Iterator iter_prox = ccn.getClientProxies().iterator();
-      if (iter_prox.hasNext()) {
-        CacheClientProxy proxy = (CacheClientProxy) iter_prox.next();
-        assertFalse("Dispatcher on secondary should not be alive",
-            proxy._messageDispatcher.isAlive());
-      }
+    final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
 
-    } catch (Exception ex) {
-      Assert.fail("while setting verifyDispatcherIsNotAlive  ", ex);
+    await().atMost(180, SECONDS).pollInterval(1, SECONDS).until(() -> ccn.getClientProxies().size(),
+        greaterThan(0));
+
+    Iterator<CacheClientProxy> cacheClientProxyIterator = ccn.getClientProxies().iterator();
+    if (cacheClientProxyIterator.hasNext()) {
+      CacheClientProxy proxy = cacheClientProxyIterator.next();
+      assertThat(proxy._messageDispatcher.isAlive())
+          .describedAs("Dispatcher on secondary should not be alive").isFalse();
     }
   }
 
-  public static void verifyRedundantServersContain(final String server) {
-    WaitCriterion wc = new WaitCriterion() {
-      public boolean done() {
-        return pool.getRedundantNames().contains(server);
-      }
+  static void verifyRedundantServersContain(final String server) {
 
-      public String description() {
-        return "Redundant servers (" + pool.getRedundantNames() + ") does not contain " + server;
-      }
-    };
-    Wait.waitForCriterion(wc, 60 * 1000, 2000, true);
+    await()
+        .atMost(180, SECONDS).pollInterval(2, SECONDS)
+        .until(() -> pool.getRedundantNames().contains(server));
   }
 
-  public static void verifyLiveAndRedundantServers(final int liveServers,
+  /**
+   *
+   * @param connectedServers is the expected number of connected servers (0 or more)
```

Comment:
This should say "1 or more". The big learning here was that it's not possible to test for zero connected servers via the API (calling through the API in that case results in the `NoSubscriptionServersAvailableException` exception).